features/admin.actions.v1/pages/action-configuration-page.tsx

```diff
@@ -0,0 +1,428 @@
+            case ActionsConstants.ACTION_TYPES.PRE_UPDATE_PASSWORD.getUrlPath():
+                return ActionsConstants.ACTION_TYPES.PRE_UPDATE_PASSWORD.getApiPath();
+            case ActionsConstants.ACTION_TYPES.PRE_UPDATE_PROFILE.getUrlPath():
+                return ActionsConstants.ACTION_TYPES.PRE_UPDATE_PROFILE.getApiPath();
+            case ActionsConstants.ACTION_TYPES.PRE_REGISTRATION.getUrlPath():
+                return ActionsConstants.ACTION_TYPES.PRE_REGISTRATION.getApiPath();
+            default:
+                return null;
+        }
+    }, [ history.location.pathname ]);
+
+    const {
+        data: actions,
+        error: actionsFetchRequestError,
+        isLoading: isActionsLoading,
+        mutate: mutateActions
+    } = useGetActionsByType(actionTypeApiPath);
+
+    const isLoading: boolean = isActionsLoading || !actions || !Array.isArray(actions);
+
+    const actionInitialValues: ActionConfigFormPropertyInterface = useMemo(() => {
+        if (actions) {
+            return {
+                authenticationType: actions[0]?.endpoint?.authentication?.type.toString(),
+                endpointUri: actions[0]?.endpoint?.uri,
+                id: actions[0]?.id,
+                name: actions[0]?.name
+            };
+        } else {
+            return null;
+        }
+    }, [ actions ]);
+
+    useEffect(() => {
+        if (actions?.length >= 1) {
+            setShowCreateForm(false);
+            setIsActive(actions[0]?.status.toString() === ActionsConstants.ACTIVE_STATUS);
+        } else {
+            setShowCreateForm(true);
+        }
+    }, [ actions ]);
+
+    /**
+     * The following useEffect is used to handle if any error occurs while fetching the Action.
+     */
+    useEffect(() => {
+        if (actionsFetchRequestError && !isActionsLoading) {
+            if (actionsFetchRequestError.response && actionsFetchRequestError.response.data
+                && actionsFetchRequestError.response.data.description) {
+                dispatch(
+                    addAlert<AlertInterface>({
+                        description: t("console:manage.features.actions.notification.error.fetch.description",
+                            { description: actionsFetchRequestError.response.data.description }),
+                        level: AlertLevels.ERROR,
+                        message: t("console:manage.features.actions.notification.error.fetch.message")
+                    })
+                );
+            } else {
+                // Generic error message
+                dispatch(
+                    addAlert<AlertInterface>({
+                        description: t("console:manage.features.actions.notification.genericError.fetch.description"),
+                        level: AlertLevels.ERROR,
+                        message: t("console:manage.features.actions.notification.genericError.fetch.message")
+                    })
+                );
+            }
+        }
+    }, [ ]);
```

Comment:
1. Add `isActionsLoading` and `actionsFetchRequestError` to the dependency array.
2. Improve readability.
```suggestion
    useEffect(() => {
        if (isActionsLoading || !actionsFetchRequestError {
            return;
        }
        
            if (actionsFetchRequestError.response?.data?.description) {
                dispatch(
                    addAlert<AlertInterface>({
                        description: t("console:manage.features.actions.notification.error.fetch.description",
                            { description: actionsFetchRequestError.response.data.description }),
                        level: AlertLevels.ERROR,
                        message: t("console:manage.features.actions.notification.error.fetch.message")
                    })
                );
            } else {
                dispatch(
                    addAlert<AlertInterface>({
                        description: t("console:manage.features.actions.notification.genericError.fetch.description"),
                        level: AlertLevels.ERROR,
                        message: t("console:manage.features.actions.notification.genericError.fetch.message")
                    })
                );
            
        }
    }, [ isActionsLoading, actionsFetchRequestError ]);
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/6722/commits/5e3564e0797c09f4f20b71837b928bc9fa57c847